### ISSUE TITLE
How to display flops and regularizer_loss in tensorboard

### ISSUE THREAD

**Issue by Xiaocong6** [google-research/morph-net#40]
Hi, thanks for open souring.
I added these two lines of code，“tf.summary.scalar('RegularizationLoss', regularizer_loss)
                    tf.summary.scalar(network_regularizer.cost_name, network_regularizer.get_cost())”
but I cannot observe them in tensorboard. 
Does it still require some other operations?
Thanks for the answer.

**Comment by Xiaocong6** [comment]
Is this correct？
sess.run([train_step, cross_entropy,regularizer_loss,network_regularizer.get_cost()]

**Comment by pkch** [comment]
You would also need to save the summaries to the event file; this is not special to MorphNet, it's the same for any other tensorflow model. For an example, see https://www.tensorflow.org/guide/summaries_and_tensorboard#serializing_the_data, note the use of `FileWriter` and `tf.summary.merge_all()`.

**Comment by Xiaocong6** [comment]
Thanks! I have solved it. I want to ask another question~
If I want to observe the FLOPS in tensorboard as you shown, do I need to replace the network structure with the learned structure during training. Why the FLOPS I observed are straight lines, maybe this due to my parameter-regularization_strength is set too small, resulting in no change in network structure?
In addition, is RegularizationLoss gradually reduced in training?
Thanks a lot!

**Comment by ayp-google** [comment]
When observing FLOPs in Tensorboard, you do not need to use the learned structure.  First, try using a stronger regularization strength.  Second, you should plot a histogram of gamma and check that the value of gamma_threshold separates the alive and dead values.

Basically, as MorphNet trains it will recalculate the FLOP cost of the network by ignoring the dead channels (but without changing the actual network).  During this process, you might observe the regularization loss (i.e. gamma values) go down but the FLOP cost stays constant.  This means that the gamma values are reduced but are still larger than the threshold and considered alive.  The main causes of this are regularizer strength too weak or threshold too small.

After training with MorphNet, you can extract the learned structure that will have the FLOP cost as reported in Tensorboard.

**Comment by Xiaocong6** [comment]
Thanks.I got it.You are so kind.
